lockfile/src/cargo.rs

```diff
@@ -0,0 +1,164 @@
+use std::ffi::OsStr;
+use std::path::Path;
+
+use phylum_types::types::package::{PackageDescriptor, PackageType};
+use serde::Deserialize;
+
+use crate::{Parse, ParseResult};
+
+pub struct Cargo;
+
+#[derive(Deserialize, Debug, Clone)]
+struct CargoLock {
+    #[serde(rename = "package")]
+    packages: Vec<Package>,
+}
+
+#[derive(Deserialize, Debug, Clone)]
+struct Package {
+    name: String,
+    version: String,
+    source: Option<String>,
+}
+
+impl Parse for Cargo {
+    /// Parses `cargo.lock` files into a vec of packages
+    fn parse(&self, data: &str) -> ParseResult {
+        let mut lock: CargoLock = toml::from_str(data)?;
+        Ok(lock
+            .packages
+            .drain(..)
+            .filter_map(|package| {
+                package.source.as_ref().map(|_| PackageDescriptor::from(package.clone()))
```

Comment:
This looks rather wonky, especially with the `clone`. Seems like we just want `try_from` maybe?

Reply:
Fixed.